presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/decoders/ArrayDecoder.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.elasticsearch.decoders;
+
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.block.BlockBuilder;
+import org.elasticsearch.search.SearchHit;
+
+import java.util.List;
+import java.util.function.Supplier;
+
+import static io.prestosql.spi.StandardErrorCode.TYPE_MISMATCH;
+
+public class ArrayDecoder
+        implements Decoder
+{
+    private final Decoder elementDecoder;
+
+    public ArrayDecoder(Decoder elementDecoder)
+    {
+        this.elementDecoder = elementDecoder;
+    }
+
+    @Override
+    public void decode(SearchHit hit, Supplier<Object> getter, BlockBuilder output)
+    {
+        Object data = getter.get();
+
+        if (data == null) {
+            output.appendNull();
+        }
+        else if (data instanceof List) {
+            final BlockBuilder array = output.beginBlockEntry();
+            ((List) data).forEach(element -> elementDecoder.decode(hit, () -> element, array));
+            output.closeEntry();
+        }
+        else {
+            throw new PrestoException(TYPE_MISMATCH, "Expected list of primitives for ARRAY field");
```

Comment:
Technically, it could be composite objects (e.g., documents), so I'd rephrase as "... list of elements ..."